pyqt_sql_demo/connection/model.py

```diff
@@ -83,11 +112,22 @@ def execute(self, query):
             # Fetch first row
             self.beginResetModel()
             logging.debug("fetcher first row")
-            self._column_count = len(first_row)
+            if self.cur.description:
+                self._headers = [h[0] for h in self.cur.description]
+                self._column_count = len(self._headers)
+            else:
+                self._column_count = len(first_row)
+                self._headers = first_row.keys()
+
             logging.debug(f"column count: {self._column_count}")
-            self._headers = first_row.keys()
             logging.debug(f"headers: {self._headers}")
-            self._data = [first_row]
+
+            first_row_as__list = [first_row]
```

Comment:
Double underscore here - probably a typo. If not, then an unusual practice in Python to use double underscore in the middle of the name.